src/diff.jl

```diff
@@ -37,3 +41,19 @@ differentiate(p::RationalPoly, x::PolyVar) = (differentiate(p.num, x) * p.den -
 function differentiate{C}(p::PolyType{C}, xs::Vector{PolyVar{C}})
     [differentiate(p, x) for x in xs]
 end
+
+# In Julia v0.5, it returns Any
+Base.promote_op{C}(differentiate, ::Type{PolyVar{C}}, ::Type{PolyVar{C}}) = Term{true, Int}
```

Comment:
is it intended for ``Base.promote_op`` to dispatch on the function ``differentiate``? Right now it's just the name of an argument (might as well be ``foo``).